### PR TITLE
Add variables about wrapping to config

### PR DESCRIPTION
**Pull request by klshrinidhi**
Adding variables to let other projects built on OpenSim know if OpenSim was built with/without wrapping and get relevant files.

This is in progress with opensim-gui PR [#33](https://github.com/opensim-org/opensim-gui/pull/33).

### CODE REVIEW

File: cmake/OpenSimConfig.cmake.in [hunk 1]
```diff
@@ -2,13 +2,23 @@
...
+#                                   Suitable for link_directories().
+#   @CMAKE_PROJECT_NAME@_INCLUDE_DIRS -- Suitable for include_directories().
+#   @CMAKE_PROJECT_NAME@_LIBRARIES -- Suitable for target_link_libraries().
+#   @CMAKE_PROJECT_NAME@_JAVA_WRAPPING -- Boolean indicating whether or not
+#                                         OpenSim was built to include Java
+#                                         wrapping.
+#   @CMAKE_PROJECT_NAME@_PYTHON_WRAPPING -- Boolean indicating whether or not
+#                                           OpenSim was built to include Python
+#                                           wrapping.
+#   @CMAKE_PROJECT_NAME@_JAR_FILE -- Path to jar file..
```
Comment (chrisdembia) [inline cmake/OpenSimConfig.cmake.in]: One period only?

File: cmake/OpenSimConfig.cmake.in [hunk 2]
```diff
@@ -54,3 +64,14 @@ include("@PACKAGE_OPENSIM_INSTALL_CMAKEDIR@/@CMAKE_PROJECT_NAME@Targets.cmake")
...
+
+# Python/Java Wrapping
+#---------------------
+set(@CMAKE_PROJECT_NAME@_PYTHON_WRAPPING @BUILD_PYTHON_WRAPPING@)
+
+set(@CMAKE_PROJECT_NAME@_JAVA_WRAPPING @BUILD_JAVA_WRAPPING@)
+set(@CMAKE_PROJECT_NAME@_JAVA_FILES_DIR 
+   @CMAKE_INSTALL_PREFIX@/@OPENSIM_INSTALL_JAVAJARDIR@/org)
+set(@CMAKE_PROJECT_NAME@_JAR_FILE 
+    @CMAKE_INSTALL_PREFIX@/@OPENSIM_INSTALL_JAVAJARDIR@/@SWIG_JAVA_JAR_NAME@)
```
Comment (chrisdembia) [inline cmake/OpenSimConfig.cmake.in]: To make the OpenSim installation relocatable, you cannot use `@CMAKE_INSTALL_PREFIX@` in the Config file. Look at how the other variables that contain install prefix are handled (e.g., `PACKAGE_OPENSIM_INSTALL_CMAKEDIR`). For documentation on how to do this, see https://github.com/opensim-org/opensim-core/blob/master/cmake/CMakeLists.txt#L14 and https://cmake.org/cmake/help/v3.0/module/CMakePackageConfigHelpers.html.

Basically, change these lines to use `@PACKAGE_OPENSIM_INSTALL_JAVAJARDIR@` and put `OPENSIM_INSTALL_JAVAJARDIR` in `cmake/CMakeLists.txt` as one of the `PATH_VARS` to `configure_package_config_file()`.

**Comment by chrisdembia** [comment]
After addressing the two comments, feel free to merge.

**Comment by klshrinidhi** [comment]
Thanks. It is not ready for review yet. There may be more changes.
Will make the changes you suggest. :+1:

**Comment by aymanhab** [comment]
@klshrinidhi Once PR #906 is in, I'll merge master into this branch and test an integrated environment including the corresponding opensim-gui PR.

**Comment by aymanhab** [comment]
@klshrinidhi Did you take care of @chrisdembia notes/observations regarding CMake? Thanks

**Comment by klshrinidhi** [comment]
I will make the changes.
Did it work on your machine ?

**Comment by aymanhab** [comment]
This now has other fixes to make the GUI work with opensim-core. I'd like to get it in as soon as convenient to automate GUI integration and avoid frequent breakages. Can you review @chrisdembia ? Thanks

File: Bindings/Java/swig/java_common.i [hunk 3]
```diff
@@ -220,6 +233,17 @@ SWIG_JAVABODY_PROXY(public, public, SWIGTYPE)
...
+	static OpenSim::Array<std::string> getFunctionClassNames() {
+		  OpenSim::Array<std::string> availableClassNames;
+		  ArrayPtrs<OpenSim::Function> rArray;
+		  Object::getRegisteredObjectsOfGivenType<OpenSim::Function>(rArray);
+		  for (int i=0;i<rArray.size(); i++)
+			availableClassNames.append(rArray[i]->getConcreteClassName());
+		  
+		  return availableClassNames;
+	}
+}
```
Comment (chrisdembia) [inline Bindings/Java/swig/java_common.i]: Do you need this twice? It appears above also.

File: cmake/OpenSimConfig.cmake.in (hunk 2 above)
Comment (chrisdembia) [inline cmake/OpenSimConfig.cmake.in]: This was not addressed yet.

**Comment by chrisdembia** [comment]
> Can you review @chrisdembia ?

Done reviewing. I made two comments.

File: Bindings/Java/swig/java_common.i (hunk 3 above)
Comment (aymanhab) [inline Bindings/Java/swig/java_common.i]: Nope, I think I added it earlier but was on a different machine so diffing showed it only once. Thanks for the find @chrisdembia

Comment (aymanhab) [inline Bindings/Java/swig/java_common.i]: Fixed in the latest commit. Thanks @chrisdembia

**Comment by klshrinidhi** [comment]
* Fixed typo.
* Fixed CMake path variables in config file as per Chris's comments.

**Comment by chrisdembia** [comment]
Looks good to me!

**Comment by klshrinidhi** [comment]
This can probably be merged without waiting for CI because it is not going to help test changes I made anyway.

**Comment by chrisdembia** [comment]
Good point. It's going in.

**Comment by aymanhab** [comment]
Awesome, thanks much @chrisdembia and @klshrinidhi
